apps/extension/src/core/views/ActivityItem.tsx

```diff
@@ -1,37 +1,21 @@
-import {
-  isMediaExtension,
-  isMediaHosted,
-  isMediaOutput,
-  isPromptInput
-} from "window.ai"
+import { isMediaHosted, isMediaOutput, isPromptInput } from "window.ai"
 
 import { Logo } from "~core/components/pure/Logo"
 import { Text } from "~core/components/pure/Text"
 import { originManager } from "~core/managers/origin"
 import type { Transaction } from "~core/managers/transaction"
 import { transactionManager } from "~core/managers/transaction"
-import { formatDate } from "~core/utils/utils"
+import { extractExtensionFromURL, formatDate } from "~core/utils/utils"
 
 function createMediaDownloadLinks({ input, outputs }: Transaction) {
-  if (!outputs) {
-    return null
-  }
-  if(!outputs.every(isMediaHosted)){
-    return "Due to storage constraints, local media generations are currently not downloadable here, but may be soon."
-  }
-  if (!isPromptInput(input)) {
+  if (!outputs || !outputs.every(isMediaOutput) || !isPromptInput(input) ||!outputs.every(isMediaHosted)) {
     return null
   }
   return outputs.map((output, index) => {
-    // defaults to no extension
-    let extension: string = ""
-    if (output.url) {
-      const base = output.url.split("?")[0]
-      const extensionFromURL = base.split(".").pop()
-      if (extensionFromURL && isMediaExtension(extensionFromURL)) {
-        extension = extensionFromURL
-      }
-    }
+    // typescript seems to need this check here, although the isMediaHosted check above should make it redundant
+    if (!output.url) return null
```

Comment:
convention: return statements shoudl be on their own lines, to show clearly when control flow is breaking out of the current function